async-query-core/src/test/java/org/opensearch/sql/spark/asyncquery/AsyncQueryCoreIntegTest.java

```diff
@@ -519,10 +608,27 @@ private void givenFlintIndexMetadataExists(String indexName) {
                     .appId(APPLICATION_ID)
                     .jobId(JOB_ID)
                     .opensearchIndexName(indexName)
-                    .flintIndexOptions(new FlintIndexOptions())
+                    .flintIndexOptions(flintIndexOptions)
                     .build()));
   }
 
+  // Overload method for default FlintIndexOptions usage
+  private void givenFlintIndexMetadataExists(String indexName) {
+    givenFlintIndexMetadataExists(indexName, new FlintIndexOptions());
+  }
+
+  // Method to set up FlintIndexMetadata with external scheduler
```

Comment:
nit: This comment looks redundant. Same for below method. (comment is not needed if the method name is well written. redundant comment would just introduce visual noise, and could become inconsistent in the future.)

Ref: https://medium.com/codex/clean-code-comments-833e11a706dc